presto-main/src/test/java/com/facebook/presto/memory/LowMemoryKillerTestingUtils.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.memory;
+
+import com.facebook.presto.client.NodeVersion;
+import com.facebook.presto.metadata.PrestoNode;
+import com.facebook.presto.spi.Node;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.memory.MemoryPoolId;
+import com.facebook.presto.spi.memory.MemoryPoolInfo;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.DataSize;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.memory.LocalMemoryManager.GENERAL_POOL;
+import static com.facebook.presto.memory.LocalMemoryManager.RESERVED_POOL;
+import static io.airlift.units.DataSize.Unit.BYTE;
+
+public class LowMemoryKillerTestingUtils
+{
+    private LowMemoryKillerTestingUtils() {}
+
+    static List<MemoryInfo> toNodeMemoryInfoList(long maxReservedPoolBytes, long maxGeneralPoolBytes, String reservedQuery, Map<String, Map<String, Long>> queries)
+    {
+        Map<Node, NodeReservation> nodeReservations = new HashMap<>();
+
+        for (Map.Entry<String, Map<String, Long>> entry : queries.entrySet()) {
+            QueryId queryId = new QueryId(entry.getKey());
+            Map<String, Long> reservationByNode = entry.getValue();
+
+            for (Map.Entry<String, Long> nodeEntry : reservationByNode.entrySet()) {
+                PrestoNode node = new PrestoNode(nodeEntry.getKey(), URI.create("http://localhost"), new NodeVersion("version"), false);
+                long bytes = nodeEntry.getValue();
+                if (bytes == 0) {
+                    continue;
+                }
+                if (reservedQuery.equals(entry.getKey())) {
+                    nodeReservations.computeIfAbsent(node, ignored -> new NodeReservation()).getReserved().add(queryId, bytes);
+                }
+                else {
+                    nodeReservations.computeIfAbsent(node, ignored -> new NodeReservation()).getGeneral().add(queryId, bytes);
```

Comment:
These two branches are a bit redundant

Reply:
I tried a few things and can't find a better way to write this though.